providers/default/ciphers/cipher_rc2.c

```diff
@@ -51,6 +83,45 @@ static int rc2_get_ctx_params(void *vctx, OSSL_PARAM params[])
         ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_SET_PARAMETER);
         return 0;
     }
+    p = OSSL_PARAM_locate(params, OSSL_CIPHER_PARAM_ALG_ID);
+    if (p != NULL) {
+        long num;
+        int i;
+        ASN1_TYPE *type;
+        unsigned char *d = p->data;
+        unsigned char **dd = d == NULL ? NULL : &d;
+
+        if (p->data_type != OSSL_PARAM_OCTET_STRING) {
+            ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_SET_PARAMETER);
+            return 0;
+        }
+        if ((type = ASN1_TYPE_new()) == NULL) {
+            ERR_raise(ERR_LIB_PROV, ERR_R_MALLOC_FAILURE);
+            return 0;
+        }
+
+        /* Is this the original IV or the running IV? */
+        num = rc2_keybits_to_magic(ctx->key_bits);
+        if (!ASN1_TYPE_set_int_octetstring(type, num,
+                                           ctx->base.iv, ctx->base.ivlen)) {
```

Comment:
This looks really dangerous to me.. it was the original iv in the old code.
There is only one iv stored in the providers ctx currently (which gets changed by cipher updates for some modes)
So we either need to
1) store the oiv as well as the iv in the provider base ctx (it is currently stored in the EVP's ctx)
2) pass the EVP's original iv down as a parameter to this function.

Reply:
I think the provider side should store the original iv

Reply:
Ok.. I will add a PR.

Reply:
Question is if that should happen in *all* implementations, or just the ones that have this custom treatment...  the latter is certainly easier!

Reply:
I will do it the first way.. That is more future proof.

Reply:
Awrightie then!